docs/protocol/on-chain/data-structures.mdx

```diff
@@ -0,0 +1,208 @@
+- `allocations`: Array of allocations defining how funds are distributed.
+- `sigs`: Array of signatures, one per <Tooltip content="An entity (identified by a wallet address) that is part of a channel.">participant</Tooltip>. Order corresponds to participants array in Channel.
+
+## Allocation
+
+Specifies how a particular amount of a token should be allocated.
+
+```solidity
+struct Allocation {
+    address destination;  // Recipient of funds
+    address token;        // ERC-20 token address
+    uint256 amount;       // Token amount in smallest unit
+}
+```
+
+**Fields**:
+
+- `destination`: Address that will receive the funds when channel closes.
+- `token`: Contract address of the ERC-20 token (or zero address for native currency).
+- `amount`: Amount in the token's smallest unit (wei for ETH, considering decimals for ERC-20).
+
+## Signature
+
+ECDSA signature components.
+
+```solidity
+struct Signature {
+    uint8 v;      // Recovery identifier
+    bytes32 r;    // First 32 bytes of signature
+    bytes32 s;    // Second 32 bytes of signature
+}
+```
+
+Signatures follow the secp256k1 curve standard used in Ethereum. The protocol is chain-agnostic and signatures are computed over the raw stateHash without EIP-191 prefix.
+
+:::caution Chain-Agnostic Signatures
+Signatures in Nitrolite are computed over raw stateHash without EIP-191 prefix, ensuring chain-agnostic compatibility.
+:::
+
+## Amount
+
+Represents a quantity of a specific token.
+
+```solidity
+struct Amount {
+    address token;    // ERC-20 token address
+    uint256 amount;   // Token amount
+}
+```
+
+## Channel Status
+
+Enum representing the lifecycle stage of a channel.
+
+```solidity
+enum Status {
+    VOID,      // Channel does not exist
+    INITIAL,   // Creation in progress, awaiting all participants
+    ACTIVE,    // Fully funded and operational
+    DISPUTE,   // Challenge period active
+    FINAL      // Ready to be closed and deleted
+}
+```
+
+**State Transitions**:
+
+```mermaid
+stateDiagram-v2
+    [*] --> VOID
+    VOID --> INITIAL: create()
+    INITIAL --> ACTIVE: join() (all participants)
+    ACTIVE --> DISPUTE: challenge()
+    ACTIVE --> FINAL: close() (cooperative)
+    DISPUTE --> ACTIVE: checkpoint() (newer state)
+    DISPUTE --> FINAL: challenge period expires
+    FINAL --> [*]
+    
+    note right of ACTIVE: Operational state<br/>Off-chain updates occur here
+    note right of DISPUTE: Challenge period active<br/>Parties can submit newer states
+```
+
+## Protocol Constants
+
+```solidity
+constant uint256 CLIENT_IDX = 0;   // Client/Creator participant index
+constant uint256 SERVER_IDX = 1;   // Server/Clearnode participant index
+constant uint256 PART_NUM = 2;     // Number of participants (always 2)
+
+constant uint16 CHANOPEN = 7877;   // 0x1EC5 - Magic number for funding state
+constant uint16 CHANCLOSE = 7879;  // 0x1EC7 - Magic number for closing state
```

Comment:
As said before, magic numbers are deprecated.
Instead, Custody defines the following additional constants:

```sol
uint256 public constant MIN_CHALLENGE_PERIOD = 1 hours;

bytes32 public constant CHALLENGE_STATE_TYPEHASH = keccak256(
        "AllowChallengeStateHash(bytes32 channelId,uint8 intent,uint256 version,bytes data,Allocation[] allocations)Allocation(address destination,address token,uint256 amount)"
    );
```